include/envoy/router/router.h

```diff
@@ -32,6 +32,51 @@ class RedirectEntry {
 };
 
 /**
+ * CorsPolicy for Route and VirtualHost.
+ */
+class CorsPolicy {
+public:
+  virtual ~CorsPolicy() {}
+
+  /**
+   * @return std::string access-control-allow-origin value.
+   */
+  virtual const std::list<std::string>& allowOrigin() const PURE;
+
+  /**
+   * @return std::string access-control-allow-methods value.
+   */
+  virtual const std::string& allowMethods() const PURE;
```

Comment:
Any particular reason that we are using a list for origins but not elsewhere? (I'm not super familiar with CORS, just curious)

Reply:
The origins I check during the requests but the other fields I return verbatim.
In section 6.1 of https://www.w3.org/TR/cors/ I interpret the ```note: Always matching is acceptable since the list of methods can be unbounded.```
to mean we don't have to check the headers/methods. 
What do you think?

Reply:
Sure sounds good.